dashboard/settings.cpp

```diff
@@ -529,5 +548,26 @@ bool Settings::steamvr_lh() const
 	return false;
 #endif
 }
+bool Settings::hid_forwarding() const
+{
+	// only called from the UI thread → no locking needed
+	static std::optional<bool> can_open_uinput;
+	if (can_open_uinput.has_value())
+		return can_open_uinput.value();
+
+	can_open_uinput = false;
+	constexpr std::array<const char *, 2> paths = {"/dev/uinput", "/dev/input/uinput"};
+	for (const char * p: paths)
+	{
+		int fd = ::open(p, O_WRONLY | O_NONBLOCK);
+		if (fd >= 0)
+		{
+			::close(fd);
+			can_open_uinput = true;
+			break;
+		}
+	}
```

Comment:
Can you check when running the server through systemd user service if it can open uinput?

If it can't, then we have to add the relevant config, and maybe move the "can" method to dbus to have an actual value.